### DISCUSSION TITLE
Unable to delete an auth user

### DISCUSSION THREAD

**Discussion by cdedreuille** [supabase/supabase#1971] [Questions]
I'm trying to delete an auth user from an API created on Nextjs on `/api/delete-user` with this code:

```
const { error: errorDeleteAuth } = await supabase.auth.api.deleteUser(
  user.id,
  process.env.SUPABASE_SERVICE_ROLE_KEY
);
if (errorDeleteAuth) throw new Error(errorDeleteAuth.message);
```

But I get this error: `supabase.auth.api.deleteUser is not a function`

I know that my client is working well as I'm doing other things before that but I can't figure out why it says that it's not a function.

**Comment by easy-street** [comment]
perhaps the `@supabase/supabase-js` dependency is a version released prior to the addition of `deleteUser`? `deleteUser` was added fairly recently; available in [`@supabase/supabase-js` `v1.11.11`](https://github.com/supabase/supabase-js/releases/tag/v1.11.11) or later.

check the project's installed version:
```
// with yarn
yarn list --pattern=@supabase/supabase-js --depth=0

// with npm
npm ls @supabase/supabase-js --depth=0
```
upgrade:
```
// with yarn
yarn upgrade @supabase/supabase-js --latest

// with npm
npm install @supabase/supabase-js@latest
```

**Reply by cdedreuille** [reply]
Thanks for your reply @easy-street. I updated `subabase-js` to the latest version but I still can't delete an auth user. The error message has changed though. It's now: `Database error deleting user`.

It's worth noting that I can't delete a user from the dashboard either. I get this message:

<img width="381" alt="Screenshot 2021-06-15 at 07 50 30" src="https://user-images.githubusercontent.com/1540635/122000271-d666d880-cdae-11eb-808d-ffdb99f40593.png">

I tried to change postgres user role to superuser with `ALTER USER postgres WITH SUPERUSER;` and it worked for both.

So it seems it's not a problem with my function nor supabase-js but perhaps a permission problem on the database but I can't find how to fix that without giving superuser power which I believe is not great.

**Reply by awalias** [reply]
I think on the dashboard this is executed by user `supabase_admin` so perhaps in your database `supabase_admin`doesn't have the permissions to access the auth schema? That user should have superuser access
